metrics/metrics.go

```diff
@@ -0,0 +1,245 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	prmkit "github.com/go-kit/kit/metrics/prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/client_golang/prometheus/push"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+)
+
+// Metrics labels.
+const (
+	LayerIDLabel   = "layer_id"
+	BlockIDLabel   = "block_id"
+	LastLayerLabel = "last_layer"
+)
+
+//go:generate mockgen -package=mocks -destination=./mocks/mocks.go -source=./metrics.go
+
+// Metrics defines an interface for metrics.
+type Metrics interface {
```

Comment:
don't do this please. if you really want to test metrics - implementation of the metric system should be mocked